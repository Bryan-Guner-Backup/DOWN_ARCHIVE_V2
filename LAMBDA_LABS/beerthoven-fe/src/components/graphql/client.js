import { ApolloClient } from "apollo-client";
import { InMemoryCache } from "apollo-cache-inmemory";
import { ApolloLink } from "apollo-link";
import { HttpLink } from "apollo-link-http";
import { setContext } from "apollo-link-context";
import { RetryLink } from "apollo-link-retry";

// create retry link -- since it takes some time to get
// the Okta token, this retry will cause any failed Apollo
// requests to retry
const retryLink = new RetryLink();

// create authenticated link
const authLink = setContext((_, { headers }) => {
  // return the headers to the context
  const token = localStorage.getItem("okta-token");
  return {
    headers: {
      ...headers,
      Authorization: token ? `Bearer ${token}` : "",
    },
  };
});

// apollo client setup
const link = new HttpLink({
  uri: "https://apollo.beerthoven.dev",
});

// create the client
const client = new ApolloClient({
  cache: new InMemoryCache(),
  link: ApolloLink.from([retryLink, authLink, link]),
});

export default client;
