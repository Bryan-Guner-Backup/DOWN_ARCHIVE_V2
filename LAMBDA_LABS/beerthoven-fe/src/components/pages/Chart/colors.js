export const bgColor= [
    'rgba(255, 99, 132, 0.2)',
    'rgba(54, 162, 235, 0.2)',
    'rgba(255, 206, 86, 0.2)',
    'rgba(75, 192, 192, 0.2)',
    'rgba(153, 102, 255, 0.2)',
    'rgba(255, 159, 64, 0.2)',
    'rgba(241, 130, 141,0.2)',
    'rgba(220, 198, 224, 0.2)',
    'rgba(213, 184, 255, 0.2)',
    'rgba(210, 82, 127, 0.2)',
    'rgba(255, 236, 139, 0.2)',
    'rgba(63, 195, 128, 0.2)',
    'rgba(228, 120, 51, 0.2)',
    'rgba(34, 167, 240, 0.2)',
    'rgba(155, 89, 182, 0.2)',
    'rgba(214, 69, 65, 0.2)',
    'rgba(243, 156, 18, 0.2)',
    'rgba(242, 120, 75, 0.2)',
    'rgba(54, 215, 183, 0.2)',
    'rgba(135, 211, 124, 0.2)',
    'rgba(107, 185, 240, 0.2)',
    'rgba(77, 19, 209, 0.2)', 
    'rgba(219, 10, 91, 0.2)',
    'rgba(241, 169, 160, 0.2)',
    'rgba(244, 208, 63, 0.2)',
    'rgba(22, 160, 133, 0.2)',
    'rgba(142, 68, 173, 0.2)',
    'rgba(115, 101, 152, 0.2)',
    'rgba(236, 100, 75, 0.2)',
    'rgba(240, 52, 52, 0.2)',
    'rgba(134, 226, 213, 0.2)',
    'rgba(30, 139, 195, 0.2)',
    'rgba(1, 50, 67, 0.2)',
    'rgba(255, 99, 132, 0.2)',
    'rgba(54, 162, 235, 0.2)',
    'rgba(255, 206, 86, 0.2)',
    'rgba(75, 192, 192, 0.2)',
    'rgba(153, 102, 255, 0.2)',
    'rgba(255, 159, 64, 0.2)',
    'rgba(241, 130, 141,0.2)',
    'rgba(220, 198, 224, 0.2)',
    'rgba(213, 184, 255, 0.2)',
    'rgba(210, 82, 127, 0.2)',
    'rgba(255, 236, 139, 0.2)',
    'rgba(63, 195, 128, 0.2)',
    'rgba(228, 120, 51, 0.2)',
    'rgba(34, 167, 240, 0.2)',
    'rgba(155, 89, 182, 0.2)',
    'rgba(214, 69, 65, 0.2)',
    'rgba(243, 156, 18, 0.2)',
    'rgba(242, 120, 75, 0.2)',
    'rgba(54, 215, 183, 0.2)',
    'rgba(135, 211, 124, 0.2)',
    'rgba(107, 185, 240, 0.2)',
    'rgba(77, 19, 209, 0.2)', 
    'rgba(219, 10, 91, 0.2)',
    'rgba(241, 169, 160, 0.2)',
    'rgba(244, 208, 63, 0.2)',
    'rgba(22, 160, 133, 0.2)',
    'rgba(142, 68, 173, 0.2)',
    'rgba(115, 101, 152, 0.2)',
    'rgba(236, 100, 75, 0.2)',
    'rgba(240, 52, 52, 0.2)',
    'rgba(134, 226, 213, 0.2)',
    'rgba(30, 139, 195, 0.2)',
    'rgba(1, 50, 67, 0.2)'

  ]

  export const bgColorBorder= [
    'rgba(255, 99, 132, 1)',
    'rgba(54, 162, 235, 1)',
    'rgba(255, 206, 86, 1)',
    'rgba(75, 192, 192, 1)',
    'rgba(153, 102, 255, 1)',
    'rgba(255, 159, 64, 1)',
    'rgba(241, 130, 141,1)',
    'rgba(220, 198, 224, 1)',
    'rgba(213, 184, 255, 1)',
    'rgba(210, 82, 127, 1)',
    'rgba(255, 236, 139, 1)',
    'rgba(63, 195, 128, 1)',
    'rgba(228, 120, 51, 1)',
    'rgba(34, 167, 240, 1)',
    'rgba(155, 89, 182, 1)',
    'rgba(214, 69, 65, 1)',
    'rgba(243, 156, 18, 1)',
    'rgba(242, 120, 75, 1)',
    'rgba(54, 215, 183, 1)',
    'rgba(135, 211, 124, 1)',
    'rgba(107, 185, 240, 1)',
    'rgba(77, 19, 209, 1)',
    'rgba(219, 10, 91, 1)',
    'rgba(241, 169, 160, 1)',
    'rgba(244, 208, 63, 1)',
    'rgba(22, 160, 133, 1)',
    'rgba(142, 68, 173, 1)',
    'rgba(115, 101, 152, 1)',
    'rgba(236, 100, 75, 1)',
    'rgba(240, 52, 52, 1)',
    'rgba(134, 226, 213, 1)',
    'rgba(30, 139, 195, 1)',
    'rgba(1, 50, 67, 1)',
    'rgba(255, 99, 132, 1)',
    'rgba(54, 162, 235, 1)',
    'rgba(255, 206, 86, 1)',
    'rgba(75, 192, 192, 1)',
    'rgba(153, 102, 255, 1)',
    'rgba(255, 159, 64, 1)',
    'rgba(241, 130, 141,1)',
    'rgba(220, 198, 224, 1)',
    'rgba(213, 184, 255, 1)',
    'rgba(210, 82, 127, 1)',
    'rgba(255, 236, 139, 1)',
    'rgba(63, 195, 128, 1)',
    'rgba(228, 120, 51, 1)',
    'rgba(34, 167, 240, 1)',
    'rgba(155, 89, 182, 1)',
    'rgba(214, 69, 65, 1)',
    'rgba(243, 156, 18, 1)',
    'rgba(242, 120, 75, 1)',
    'rgba(54, 215, 183, 1)',
    'rgba(135, 211, 124, 1)',
    'rgba(107, 185, 240, 1)',
    'rgba(77, 19, 209, 1)',
    'rgba(219, 10, 91, 1)',
    'rgba(241, 169, 160, 1)',
    'rgba(244, 208, 63, 1)',
    'rgba(22, 160, 133, 1)',
    'rgba(142, 68, 173, 1)',
    'rgba(115, 101, 152, 1)',
    'rgba(236, 100, 75, 1)',
    'rgba(240, 52, 52, 1)',
    'rgba(134, 226, 213, 1)',
    'rgba(30, 139, 195, 1)',
    'rgba(1, 50, 67, 1)'

  ]


