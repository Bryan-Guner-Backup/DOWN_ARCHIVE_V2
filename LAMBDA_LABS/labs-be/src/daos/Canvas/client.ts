import { Request, Response } from "express";
import {
  BaseClient,
  AuthTypes,
  ClientResponse,
  ClientArrayResponse,
} from "@shared/BaseClient";

export interface ICanvasClient<T> {
  get: (path: string) => ClientResponse<T> | ClientArrayResponse<T>;
}

class CanvasClient<T> extends BaseClient<T> implements ICanvasClient<T> {
  constructor() {
    const opts = {
      token: process.env.CANVAS_ACCESS_TOKEN,
      baseUrl: "https://lambdaschool.instructure.com/api/v1/",
      authType: AuthTypes.JWT,
    };

    super(opts);
  }
}

export default CanvasClient;
