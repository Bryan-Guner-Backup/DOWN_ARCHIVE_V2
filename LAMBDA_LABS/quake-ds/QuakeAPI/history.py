from flask import jsonify
from math import degrees, radians, cos, sin, asin, atan2, sqrt


def history(lat, lon, dist):
    '''Calculate the top left and bottom right coordinates at distance dist
    from coordinates lat, lon'''

    lat = radians(lat)
    lon = radians(lon)
    radius = 6371
    top_left = radians(315)
    bottom_right = radians(135)

    lat2A = asin(sin(lat) * cos(dist/radius) +
                 cos(lat) * sin(dist/radius) * cos(top_left))
    lat2B = asin(sin(lat) * cos(dist/radius) +
                 cos(lat) * sin(dist/radius) * cos(bottom_right))

    lon2A = lon + atan2(sin(top_left) * sin(dist/radius) *
                        cos(lat), cos(dist/radius) - sin(lat) * sin(lat2A))
    lon2B = lon + atan2(sin(bottom_right) * sin(dist/radius) *
                        cos(lat), cos(dist/radius) - sin(lat) * sin(lat2B))

    coordinates = {
        'lonA': degrees(lon2A),
        'latA': degrees(lat2A),
        'lonB': degrees(lon2B),
        'latB': degrees(lat2B)
    }


    return coordinates
