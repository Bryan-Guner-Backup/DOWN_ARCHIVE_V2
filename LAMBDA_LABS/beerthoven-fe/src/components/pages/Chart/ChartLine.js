import React, { useState, useEffect } from 'react';
import { useQuery } from '@apollo/react-hooks';
import { Line } from 'react-chartjs-2';
import { ALL_EVENTS } from '../../graphql/queries';
import '../../../index.css'

//TO GET LABEL NAME
//sort data by date
const sortByDate = (data) => {
  return data.sort((a, b) => {
    let dateA = new Date(a.event_date), dateB = new Date(b.event_date);
    return dateA - dateB;
  })

}
const ChartLine = () => {

  const [chartData, setChartData] = useState({})
  const { data } = useQuery(ALL_EVENTS);

  useEffect(() => {
    if (data && data.events) {

      sortByDate(data.events)

      //REMOVE DUPLICATES
      const totals = data.events.reduce(
        (totals,{ event_date, sales_net }) =>
          totals[new Date(event_date).toDateString()]
            ? { ...totals, [new Date(event_date).toDateString()]: totals[new Date(event_date).toDateString()] + sales_net }
            : { ...totals, [new Date(event_date).toDateString()]: sales_net },
        {}
      );
      const dailyTotals = Object.values(totals)

       //Extract and reformat Date string
      const dateLabel = (Object.keys(totals)).map(k => (k.split(' ').slice(1).join(' ')))

      setChartData({
        labels: dateLabel,
        datasets: [{
          label: 'sales',
          fill: 'none',
          data: dailyTotals,
          backgroundColor: 'rgba(255, 99, 132, 1)',
          borderColor: 'rgba(255, 99, 132, 1)',
          borderWidth: 1,
          pointRadius: 2,
        }]
      })
    }

  }, [data])

  return (<div>

    {data && data.events.length && <ChartLineSection chartData={chartData} />}
  </div>)
}

const ChartLineSection = ({ chartData }) => {

  const dateRange = `${chartData.labels ? chartData.labels[0] : 'na'} to ${chartData.labels ? chartData.labels[chartData.labels.length - 1] : 'na'}`
  return (

    <Line
      width={600}
      data={chartData}
      options={{
        elements: {
          line: {
            tension: 0.000001
          }
        },
        responsive: true,
        title: { text: `Net Sales from ${dateRange}`, display: true },
        scales: {
          yAxes: [{
            ticks: {
              beginAtZero: true,
              callback: function(value, index, values) {
                return '$' + value;
              }
            }
          }]
        },
      }}
    />

  )
}


export default ChartLine

