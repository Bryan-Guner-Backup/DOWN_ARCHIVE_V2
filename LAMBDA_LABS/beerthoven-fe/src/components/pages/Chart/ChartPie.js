import React, { useState, useEffect } from 'react';
import { useQuery } from '@apollo/react-hooks';
import { Line , Pie} from 'react-chartjs-2';
import { ALL_USERS } from '../../graphql/queries';
import '../../../index.css'
import { bgColor, bgColorBorder } from './colors';


//TO GET LABEL NAME
//sort data by date

const sortByDate = (data) => {
  return data.sort((a, b) => {
    let dateA = new Date(a.event_date), dateB = new Date(b.event_date);
    return dateA - dateB;
  })

}
const ChartPie = () => {

  const [chartData, setChartData] = useState({})
  const { data } = useQuery(ALL_USERS);

  useEffect(() => {
    if (data && data.users) {

    //   const sortData = sortByDate(data.events)
    //   //Extract and reformat Date string

        const userRoles= data.users.map(item=>item.role)

      //REMOVE DUPLICATES LABEL ROLES
      const roleLabel = [...new Set(userRoles)]

      //COUNT EACH USER IN EACH ROLE
    const counts = {}
      userRoles.forEach((x) => { counts[x] = (counts[x] || 0) + 1 })
      const userCount = Object.values(counts)
      setChartData({
        labels: roleLabel,
        datasets: [{
          label: 'sales',
          data: userCount,
          backgroundColor: bgColor,
          borderColor: bgColorBorder,
          borderWidth: 1,
          pointRadius: 2,
        }]
      })
    }

  }, [data])

  return (<div>
    {data && data.users.length && <ChartPieSection chartData={chartData} />}
  </div>)
}

const ChartPieSection = ({ chartData }) => {

//   const dateRange = `${chartData.labels ? chartData.labels[0] : 'na'} to ${chartData.labels ? chartData.labels[chartData.labels.length - 1] : 'na'}`
  return (
      <>
    <Pie
          data={chartData}
          options={{
            title: {
              display: true,
              text: 'Beerthoven Users- Pie Graph',
              fontSize: 20
            },
            legend: {
              display: true,
              position: 'left'
            },
          }}

        />
        </>

  )
}


export default ChartPie

