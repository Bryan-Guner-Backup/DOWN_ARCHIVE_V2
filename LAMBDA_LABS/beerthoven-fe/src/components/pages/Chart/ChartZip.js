import React, { useState, useEffect } from 'react';
import { useQuery } from '@apollo/react-hooks';
import { Doughnut } from 'react-chartjs-2';
import { ALL_PERSONS } from '../../graphql/queries';
import '../../../index.css'
import { bgColor, bgColorBorder } from './colors';



const ChartZip = () => {

  const [chartData, setChartData] = useState({})
  const { data } = useQuery(ALL_PERSONS);


  useEffect(() => {
    if (data && data.persons) {
      const people = data.persons.map(item => item.zip)
      const counts = {}
      people.forEach((x) => { counts[x] = (counts[x] || 0) + 1 })
      const pplNum = Object.values(counts)

      const countEntry = Object.fromEntries(
        Object.entries(counts).slice(0, 20)
      );


      setChartData(
        {

          labels: Object.keys(countEntry),
          datasets: [{
            label: '# of people per zip code',
            data: Object.values(countEntry),
            backgroundColor: bgColor,
            borderColor: bgColorBorder,
            borderWidth: 1
          }]
        }

      )
    }

  }, [data])


  return (<div>
    {data && data.persons.length && <ChartZipSection chartData={chartData} />}
  </div>)
}

const ChartZipSection = ({ chartData }) => {

  const dateRange = `${chartData.labels ? chartData.labels[0] : 'na'} to ${chartData.labels ? chartData.labels[chartData.labels.length - 1] : 'na'}`
  return (

    <Doughnut
      data={chartData}
      options={{
        title: {
          display: true,
          text: 'Attendee Location- Doughnut Graph',
          fontSize: 20
        },
        legend: {
          display: true,
          position: 'right', 
          align: 'center',
        }
      }}
    />
  )
}


export default ChartZip

