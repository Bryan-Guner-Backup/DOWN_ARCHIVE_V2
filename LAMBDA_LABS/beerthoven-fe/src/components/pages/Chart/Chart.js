import React, { useState, useEffect } from 'react';
import { useQuery } from '@apollo/react-hooks';
import { Pie, Doughnut, Line, Bar } from 'react-chartjs-2';
import { ALL_EVENTS } from '../../graphql/queries';
import ChartLine from './ChartLine'
import ChartPie from './ChartPie'
import ChartZip from './ChartZip'
import '../../../index.css'
import { bgColor, bgColorBorder } from './colors';

const Chart = () => {

  const [chartData, setChartData] = useState({})

  const { data } = useQuery(ALL_EVENTS);
  useEffect(() => {
    if (data && data.events) {
      const states = data.events.map(item => item.state) //replace item.state with item.zip once all data is transfered
      const stateNames = states.filter((item, index, self) => self.indexOf(item) === index)
      const counts = {}
      states.forEach((x) => { counts[x] = (counts[x] || 0) + 1 })
      const statesNum = Object.values(counts)
      setChartData({
        labels: stateNames,
        datasets: [{
          label: '# of events in state',
          data: statesNum,
          backgroundColor: bgColor,
          borderColor: bgColorBorder,
          borderWidth: 1
        }]
      })
    }

  }, [data])

  return (<div>

    {data && data.events.length && (
      <>
        <ChartSection chartData={chartData} list={data.events} />
        <ChartLine />

      </>
    )}
  </div>)
}

const ChartSection = ({ chartData }) => {
  return (
    <div className="container">
      <div className="chart-container">
        <Pie
          data={chartData}
          options={{
            title: {
              display: true,
              text: 'Event Location - Pie Graph',
              fontSize: 20
            },
            legend: {
              display: true,
              position: 'left'
            },
          }}

        />
      </div>

      <div className="chart-container">
        <ChartZip />
      </div>

      <div className="chart-container">
        <Bar data={chartData} 
        options={{ 
          title: { text: "Event location - Bar Chart", 
                    display: true }, 
          scales: {
                      yAxes: [
                        { ticks: { beginAtZero: true }}
                      ]
                  }
        }}
           />
      </div>

      <div className="chart-container">


        <Line
          data={chartData}
          options={{
            responsive: true,
            title: { text: "Event location - Line Chart", display: true },
            scales: {
              yAxes: [
                {
                  ticks: {
                    autoSkip: true,
                    maxTicksLimit: 10,
                    beginAtZero: true
                  },
                  gridLines: { display: false }
                }
              ],
              xAxes: [{ gridLines: { display: false } }]
            }
          }}
        />
      </div>

      <div className="chart-container">

      <ChartPie/>
 
       </div>

    </div>

    
  )
}


export default Chart