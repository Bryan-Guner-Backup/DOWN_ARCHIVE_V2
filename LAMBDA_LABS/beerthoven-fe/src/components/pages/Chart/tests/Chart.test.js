import React from 'react';
import Chart from '../Chart';
import { render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import { Router } from 'react-router'
import { createMemoryHistory } from "history";
import { ApolloProvider } from 'react-apollo';
import client from '../../../graphql/client';

const onSubmit = jest.fn()


describe(`People Form Component`, () => {
    beforeEach(cleanup);

    it(`renders a component that uses withRouter`, async () => {
        const history = createMemoryHistory()
        const route = '/some-route'
        history.push(route)
        const { container } = render(
            <ApolloProvider client={client}>
                <Router history={history}><Chart /></Router>
            </ApolloProvider>
        )

        expect(container.firstChild).toMatchSnapshot();
    })
})
