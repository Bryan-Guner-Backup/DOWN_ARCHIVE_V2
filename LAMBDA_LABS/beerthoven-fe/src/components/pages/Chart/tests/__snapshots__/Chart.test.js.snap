// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People Form Component renders a component that uses withRouter 1`] = `<div />`;
